core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1092 @@
+  ) {
+    this._quadrant = quadrant;
+    this._fractions = fractions;
+    this._axisAtStart = axisAtStart;
+    this._axisAtEnd = axisAtEnd;
+    this._averageAdded = false;
+  }
+  /** Constructor, captures the array. */
+  public static create(
+    quadrant: 1 | 2 | 3 | 4, fractions: number[] = [], axisAtStart: boolean = false, axisAtEnd: boolean = false,
+  ): QuadrantFractions {
+    return new QuadrantFractions(quadrant, fractions, axisAtStart, axisAtEnd);
+  }
+  /**
+   * Quadrant of the full ellipse containing the samples.
+   * * Quadrants are labeled proceeding in counterclockwise angular sweeps of length pi/2 starting at vector0.
+   * * For example, Quadrant 1 starts at vector0 and ends at vector90, and Quadrant 4 ends at vector0.
+   * * For purposes of angle classification, quadrants are half-open intervals, closed at their start angle,
+   * as determined by the ellipse's sweep direction.
+   */
+  public get quadrant(): 1 | 2 | 3 | 4 {
+    return this._quadrant;
+  }
+  /** Sample locations in this quadrant of the elliptical arc, as fractions of its sweep. */
+  public get fractions(): number[] {
+    return this._fractions;
+  }
+  public set fractions(f: number[]) {
+    this._fractions = f;
+  }
+  /**
+   * Whether the first fraction is the location of an ellipse axis point.
+   * * Only valid if `this.fractions.length > 0`.
+   */
+  public get axisAtStart(): boolean {
+    return this._fractions.length > 0 ? this._axisAtStart : false;
+  }
+  public set axisAtStart(onAxis: boolean) {
+    this._axisAtStart = onAxis;
+  }
+  /**
+   * Whether the last fraction is the location of an ellipse axis point.
+   * * Only valid if `this.fractions.length > 1`.
+   */
+  public get axisAtEnd(): boolean {
+    return this._fractions.length > 1 ? this._axisAtEnd : false;
+  }
+  public set axisAtEnd(onAxis: boolean) {
+    this._axisAtEnd = onAxis;
+  }
+  /**
+   * Whether the average fraction was added to produce a fractions array of length 3.
+   * * Some sample methods and elliptical arcs may not yield enough samples in a quadrant.
+   * * This flag is set if an extra (average) fraction was added to a 2-fraction array to satisfy
+   * the minimum 3-sample requirement that allows both axis/endpoints in the quadrant to be interpolated
+   * by [[EllipticalArcApproximationContext.constructCircularArcChainApproximation]].
+  */
+  public get averageAdded(): boolean {
+    return this._fractions.length === 3 ? this._averageAdded : false;
+  }
+  public set averageAdded(added: boolean) {
+    this._averageAdded = added;
+  }
+  /**
+   * Compute quadrant data for the given angles.
+   * @param radians0 first radian angle
+   * @param radians1 second radian angle
+   * @return quadrant number and start/end radian angles for the quadrant that contains both input angles, or
+   * undefined if no such quadrant.
+   * * The returned sweep is always counterclockwise: angle0 < angle1.
+   */
+  public static getQuadrantRadians(
```

Comment:
So one can pass `radian0` and `radian1` in different quadrants and then this function returns `undefined`, right? I so, please resolve this comment.

Reply:
Correct!

Reply:
Initially I though it's better to make this function `private` but then it cannot be called in this file (because caller is in another class)